genie-agent/src/main/java/com/netflix/genie/agent/cli/CliAutoConfiguration.java

```diff
@@ -18,12 +18,17 @@
 package com.netflix.genie.agent.cli;
 
 import com.netflix.genie.agent.AgentMetadata;
+import com.netflix.genie.agent.cli.logging.AgentLogManager;
+import com.netflix.genie.agent.cli.logging.AgentLogManagerLog4j2Impl;
 import com.netflix.genie.agent.execution.services.AgentHeartBeatService;
 import com.netflix.genie.agent.execution.services.AgentJobService;
 import com.netflix.genie.agent.execution.services.DownloadService;
 import com.netflix.genie.agent.execution.services.KillService;
 import com.netflix.genie.agent.execution.statemachine.JobExecutionStateMachine;
 import com.netflix.genie.proto.PingServiceGrpc;
+import org.apache.logging.log4j.LogManager;
```

Comment:
Since you made the bean conditional on a class shouldn't this direct import not be here as it will fail to start if someone (for some reason) ripped out log4j?

Reply:
I'm not 100% sure, but I don't think so, it will fail to compile if log4j2 is not in the classpath, but we for example bring in plenty of spring autoconfigurations that depend on classes not in the classpath, and there's no runtime error (example: `RedisHealthContributorAutoConfiguration`).

Beside, this is a CLI application and it's not supposed to be included as a library.

I *think* (but it may be wrong) this allows a wrapper project to provide a substitute bean for a substitute logging library.